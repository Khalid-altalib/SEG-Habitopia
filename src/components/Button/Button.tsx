import React from "react";
import { StyleSheet, TouchableOpacity, View, ViewStyle } from "react-native";

import Text from "../Text";
import { ButtonType, TextType } from "../../../types";
import { theme, useColorModeValue } from "native-base";
import { LinearGradient } from "expo-linear-gradient";

type Props = {
  style?: ViewStyle;
  isFullWidth?: boolean;
  onPress?: () => void;
  type: ButtonType;
  icon?: JSX.Element;
  children: string;
};

const Button = ({
  style,
  isFullWidth = false,
  onPress,
  type,
  icon,
  children,
}: Props) => {
  if (type == undefined) type = ButtonType.Primary;

  const styles = StyleSheet.create({
    base: {
      borderRadius: 10,
      paddingVertical: 15,
      paddingHorizontal: 40,
      alignItems: "center",
      justifyContent: "center",
    },
  });

  const typeStyles = {
    primary: {
      backgroundColors: [theme.colors.darkBlue[500], theme.colors.purple[700]],
    },
    secondary: {
      backgroundColors: useColorModeValue(
        [theme.colors.gray[500], theme.colors.gray[400]],
        [theme.colors.blueGray[600], theme.colors.blueGray[700]]
      ),
    },
  };

  const additionalStyle = ButtonType[type]
    .toString()
    .toLowerCase() as keyof typeof typeStyles;

  return (
    <View
      style={[
        {
          shadowColor: "black",
          shadowOffset: { width: 5, height: 5 },
          shadowOpacity: 0.15,
          borderRadius: 10,
        },
      ]}
    >
      <TouchableOpacity activeOpacity={0.7} onPress={onPress} style={style}>
        <LinearGradient
          colors={typeStyles[additionalStyle].backgroundColors}
          start={[0, 0]}
          end={[1, 1]}
          style={[
            styles.base,
            isFullWidth ? { width: "100%" } : { alignSelf: "flex-start" },
          ]}
        >
          <View style={{ flexDirection: "row", alignItems: "center" }}>
            <Text
              type={TextType.Button}
              style={icon == undefined ? null : { marginRight: 5 }}
            >
              {children}
            </Text>
            {icon}
          </View>
        </LinearGradient>
      </TouchableOpacity>
    </View>
  );
};

export default Button;
