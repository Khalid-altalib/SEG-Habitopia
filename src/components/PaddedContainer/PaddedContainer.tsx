import React from "react";
import { View } from "react-native";

type Props = {
  children: any;
};

/**
 * Displays children components in a container with predefined padding.
 *
 * @param props The children which are to be displayed within the container.
 * @returns The padded container with its children.
 */
const PaddedContainer = (props: Props) => {
  const PADDING = 15;

  return <View style={{ padding: PADDING }}>{props.children}</View>;
};

export default PaddedContainer;
