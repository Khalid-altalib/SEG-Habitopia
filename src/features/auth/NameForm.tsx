import React from "react";
import { Input } from "native-base";
import { useForm, Controller } from "react-hook-form";
import { AntDesign } from "@expo/vector-icons";

import { addLogInData, addSignUpData, logInUser } from "./authSlice";
import { useDispatch } from "../../app/hooks";
import Button from "../../components/Button";
import Text from "../../components/Text";
import { AuthParams, ButtonType, TextType } from "../../../types";
import { View } from "react-native";
import { NavigationProp, useNavigation } from "@react-navigation/native";
import { Button as TButton } from "native-base";

type formData = {
  name: string;
};

const NameForm = () => {
  const { control, handleSubmit } = useForm({
    defaultValues: {
      name: "",
    },
  });
  const dispatch = useDispatch();

  const onSubmit = async (data: formData) => {
    dispatch(addSignUpData(data));
    navigation.navigate("EmailAddress");
  };

  const navigation = useNavigation<NavigationProp<AuthParams>>();

  return (
    <View>
      <Text style={{ marginBottom: 5 }} type={TextType.Subtle}>
        Name
      </Text>
      <Controller
        control={control}
        render={({ field: { onChange, value } }) => (
          <Input onChangeText={onChange} value={value} size="xl" />
        )}
        name="name"
      />
      <Button
        onPress={handleSubmit(onSubmit)}
        type={ButtonType.Primary}
        style={{ marginTop: 20 }}
        isFullWidth
        icon={<AntDesign name="arrowright" size={20} color="white" />}
      >
        Continue
      </Button>
    </View>
  );
};

export default NameForm;
