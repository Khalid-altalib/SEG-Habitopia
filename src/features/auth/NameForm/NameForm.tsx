import React from "react";
import { Input } from "native-base";
import { useForm, Controller } from "react-hook-form";
import { AntDesign } from "@expo/vector-icons";

import { addLogInData, addSignUpData, logInUser } from "../authSlice";
import { useDispatch } from "../../../app/hooks";
import Button from "../../../components/Button";
import Text from "../../../components/Text";
import { AuthParams, ButtonType, TextType } from "../../../../types";
import { View } from "react-native";
import { NavigationProp, useNavigation } from "@react-navigation/native";
import { Toast } from "react-native-toast-message/lib/src/Toast";

type formData = {
  name: string;
};


/**
 * A form for accepting the user's name
 *
 * @returns The component representing the name form
 */
const NameForm = () => {
  const { control, handleSubmit } = useForm({
    defaultValues: {
      name: "",
    },
  });
  const dispatch = useDispatch();

  const onSubmit = async (data: formData) => {
    if (data.name.length < 3) {
      Toast.show({
        type: "error",
        text1: "Please ensure name is longer than 3 characters",
      });
    } else {
      dispatch(addSignUpData(data));
      navigation.navigate("EmailAddress");
    }
  };

  const navigation = useNavigation<NavigationProp<AuthParams>>();

  return (
    <View>
      <Text style={{ marginBottom: 5 }} type={TextType.Subtle}>
        Name
      </Text>
      <Controller
        control={control}
        render={({ field: { onChange, value } }) => (
          <Input
            onChangeText={onChange}
            value={value}
            size="xl"
            testID="name-input"
            autoCapitalize="none"
          />
        )}
        name="name"
      />
      <Button
        onPress={handleSubmit(onSubmit)}
        type={ButtonType.Primary}
        style={{ marginTop: 20 }}
        isFullWidth
        icon={<AntDesign name="arrowright" size={20} color="white" />}
      >
        Continue
      </Button>
    </View>
  );
};

export default NameForm;
