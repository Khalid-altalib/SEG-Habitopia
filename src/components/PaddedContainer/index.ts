import PaddedContainer from "./PaddedContainer";

export default PaddedContainer;
