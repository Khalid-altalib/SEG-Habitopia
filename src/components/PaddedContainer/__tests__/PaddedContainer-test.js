import PaddedContainer from "../PaddedContainer";
import Button from "../../Button";

import { shallow } from "enzyme";
import NameForm from "../../../features/auth/NameForm";
import MockStorage from "../../../mockStorage";

const storageCache = {};
const AsyncStorage = new MockStorage(storageCache);

jest.setMock("AsyncStorage", AsyncStorage);
it("works", () => {
  let wrap = shallow(<NameForm />);

  expect(wrap).toMatchSnapshot();
});
