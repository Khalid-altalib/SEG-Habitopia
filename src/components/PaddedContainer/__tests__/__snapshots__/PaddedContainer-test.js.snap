// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
<View
  style={
    [
      {
        "borderRadius": 10,
        "shadowColor": "black",
        "shadowOffset": {
          "height": 5,
          "width": 5,
        },
        "shadowOpacity": 0.15,
      },
    ]
  }
>
  <TouchableOpacity
    activeOpacity={0.7}
    onPress={[Function]}
  >
    <LinearGradient
      colors={
        [
          "#0077e6",
          "#7e22ce",
        ]
      }
      end={
        [
          1,
          1,
        ]
      }
      start={
        [
          0,
          0,
        ]
      }
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 10,
            "justifyContent": "center",
            "paddingHorizontal": 40,
            "paddingVertical": 15,
          },
          {
            "alignSelf": "flex-start",
          },
        ]
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={null}
          type={3}
        />
      </View>
    </LinearGradient>
  </TouchableOpacity>
</View>
`;
